src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -114,201 +165,195 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
         /// <remarks>
         /// Users can utilize <see cref="SignedHttpRequestDescriptor.AdditionalPayloadClaims"/> to create additional claim(s) and add them to the signed http request.
         /// </remarks>
-        protected virtual string CreateHttpRequestPayload(SignedHttpRequestDescriptor signedHttpRequestDescriptor, CallContext callContext)
+        protected internal virtual string CreateHttpRequestPayload(SignedHttpRequestDescriptor signedHttpRequestDescriptor, CallContext callContext)
         {
             if (signedHttpRequestDescriptor == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor));
 
-            Dictionary<string, object> payload = new Dictionary<string, object>();
+            Utf8JsonWriter writer = null;
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+                try
+                {
+                    writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
+                    writer.WriteStartObject();
+
+                    CreateHttpRequestPayload(ref writer, signedHttpRequestDescriptor);
+
+                    writer.WriteEndObject();
+                    writer.Flush();
+                    return Encoding.UTF8.GetString(memoryStream.ToArray());
```

Comment:
same comment on buffer

Reply:
Yep

Reply:
@keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f